examples/cube/main.rs

```diff
@@ -274,19 +271,27 @@ impl framework::Example for Example {
             }],
         };
 
-        let vs_module = device.create_shader_module(&wgpu::include_spirv!("shader.vert.spv"));
-        let fs_module = device.create_shader_module(&wgpu::include_spirv!("shader.frag.spv"));
+        let mut flags = wgpu::ShaderFlags::VALIDATION;
+        match adapter.get_info().backend {
+            wgpu::Backend::Metal => flags |= wgpu::ShaderFlags::EXPERIMENTAL_TRANSLATION,
+            _ => (), //TODO: https://github.com/gfx-rs/naga/issues/335
+        }
+        let shader = device.create_shader_module(&wgpu::ShaderModuleDescriptor {
+            label: None,
+            source: wgpu::ShaderSource::Wgsl(Cow::Borrowed(include_str!("shader.wgsl"))),
```

Comment:
nit: this will break the examples on the web backend for a while, though maybe we could start generating SPIR-V internally soon or temporarily keep the spv files as a fallback path for the web

Reply:
Good point! I'll hook Naga up to the web side and convert it.